### PR TITLE
Update deployment guides to use volumes

### PR DESCRIPTION
**Pull request by Joffcom**
Docker Caddy Deployment PR: https://github.com/n8n-io/n8n-docker-caddy/pull/11

### CODE REVIEW

File: docs/1-0-migration-checklist.md [hunk 1]
```diff
@@ -68,7 +70,13 @@ If permission errors appear in your n8n container logs when starting n8n, you ma
...
 ```
 
-We have also removed the Debian and RHEL images. If you were using these you need to change the image you use. This shouldn't result in any errors unless you were making a custom image based on one of those images. 
+#### Image removal
+
+We have removed the Debian and RHEL images. If you were using these you need to change the image you use. This shouldn't result in any errors unless you were making a custom image based on one of those images. 
+
+#### Entrypoint change
+
+The entrypoint for the container has changed and you no longer need to specify the n8n command, If you were previously running `n8n worker --concurrency=5` it would now be `worker --concurrency=5`
```
Comment (StarfallProjects) [inline docs/1-0-migration-checklist.md:79]:
```suggestion
The entrypoint for the container has changed and you no longer need to specify the n8n command. If you were previously running `n8n worker --concurrency=5` it's now `worker --concurrency=5`
```

File: _snippets/self-hosting/installation/docker-compose-updating.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+If you're running n8n using a Docker Compose file, follow the steps below to update n8n.
```
Comment (StarfallProjects) [inline _snippets/self-hosting/installation/docker-compose-updating.md:1]:
```suggestion
If you run n8n using a Docker Compose file, follow these steps to update n8n:
```

File: docs/hosting/installation/server-setups/digital-ocean.md [hunk 3]
```diff
@@ -65,20 +65,25 @@ cd n8n-docker-caddy
...
-### Create Docker volume
+### Create Docker volumes
 
 To persist the Caddy cache between restarts and speed up start times, create [a Docker volume](https://docs.docker.com/storage/volumes/){:target="_blank" .external-link} that Docker reuses between restarts:
 
 ```shell
 sudo docker volume create caddy_data
 ```
 
+We will also do this to create a Docker volume for the n8n data:
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/digital-ocean.md:81]:
```suggestion
Create a Docker volume for the n8n data:
```

File: docs/hosting/installation/server-setups/docker-compose.md [hunk 4]
```diff
@@ -128,15 +108,15 @@ If you are running n8n as a non-root user, add this under `volumes` for the n8n
...
 
 ### 6. Create `.env` file
 
-Create an `.env` file and change it accordingly.
+Create a `.env` file and change it accordingly.
 
 ```bash
-# Folder where data should be saved
-DATA_FOLDER=/root/n8n/
+# Replace <directory-path> with the path you will keep your Caddyfile in
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/docker-compose.md:118]:
```suggestion
# Replace <directory-path> with the path to the directory containing your Caddyfile
```

File: docs/hosting/installation/server-setups/docker-compose.md [hunk 5]
```diff
@@ -155,18 +135,30 @@ GENERIC_TIMEZONE=Europe/Berlin
 SSL_EMAIL=user@example.com
 ```
 
+### 7. Create a Caddyfile
 
-### 7. Create data folder
+Create a `Caddyfile` in the `DATA_FOLDER` path you set in the `.env` file, Make sure you replace `<subdomain>` and `<domain_name>` with the same values you set in the `.env` file.
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/docker-compose.md:140]:
```suggestion
Create a `Caddyfile` in the `DATA_FOLDER` path you set in the `.env` file. Make sure you replace `<subdomain>` and `<domain_name>` with the same values you set in the `.env` file.
```

File: docs/hosting/installation/server-setups/docker-compose.md [hunk 6]
```diff
@@ -155,18 +135,30 @@ GENERIC_TIMEZONE=Europe/Berlin
...
+      flush_interval -1
+    }
+}
+```
 
-In that folder, the database file from SQLite as well as the encryption key will be saved.
+### 7. Create the Docker volumes
 
-The folder can be created like this:
+Create the Docker volume which is defined as `n8n_data`, In this volume, the database file from SQLite as well as the encryption key will be saved.
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/docker-compose.md:152]:
```suggestion
Create the Docker volume that is defined as `n8n_data`. n8n will save the database file from SQLite and the encryption key in this volume.
```

File: docs/hosting/installation/server-setups/docker-compose.md [hunk 7]
```diff
@@ -155,18 +135,30 @@ GENERIC_TIMEZONE=Europe/Berlin
...
 
+```sh
+docker volume create n8n_data
 ```
-mkdir /root/n8n/
+
+We will also create a volume for the Caddy data, This is defined as `caddy_data`.
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/docker-compose.md:158]:
```suggestion
Create a volume for the Caddy data, This is defined as `caddy_data`.
```

File: docs/hosting/installation/server-setups/hetzner.md [hunk 8]
```diff
@@ -72,6 +71,12 @@ To persist the Caddy cache between restarts and speed up start times, create [a
 docker volume create caddy_data
 ```
 
+We will also do this to create a Docker volume for the n8n data:
```
Comment (StarfallProjects) [inline docs/hosting/installation/server-setups/hetzner.md:74]:
```suggestion
Create a Docker volume for the n8n data:
```

**Review by StarfallProjects — CHANGES_REQUESTED** [review]

File: docs/hosting/installation/server-setups/docker-compose.md [hunk 9]
```diff
@@ -60,51 +60,25 @@ IP address: <IP_OF_YOUR_SERVER>
 Create a `docker-compose.yml` file. Paste the following in the file:
 
 ```yaml
-version: "3"
+version: "3.7"
 
 services:
-  traefik:
-    image: "traefik"
```
Comment (csuermann) [inline docs/hosting/installation/server-setups/docker-compose.md:67]: I would prefer if we kept the Traefik example. No objections to providing a Caddy example in addition.

**Review by csuermann — COMMENTED** [review]
LGTM, great work, @Joffcom!
Thank you, @StarfallProjects, for your helpful suggestions!

Just one thing: I'd prefer if we (also) kept the Traefik example.

File: docs/hosting/installation/server-setups/docker-compose.md (hunk 9 above)
Comment (Joffcom) [reply docs/hosting/installation/server-setups/docker-compose.md:67]: Will change back to Traefik.

**Review by StarfallProjects — APPROVED** [review]

**Pull request merged**